master/internal/rm/agent_resource_manager.go

```diff
@@ -95,9 +96,31 @@ func (a AgentResourceManager) IsReattachEnabledForRP(ctx actor.Messenger, rpName
 	return false
 }
 
+// CheckMaxSlotsExceeded checks if the job exceeded the maximum number of slots.
+func (a AgentResourceManager) CheckMaxSlotsExceeded(
```

Comment:
one thing i can see now; on every new saas cluster, someone will see this on first job submission, every time. maybe we should make it seem less severe if we know the cluster can autoscale?

Reply:
Good point, do we have a way of being able to tell if the cluster can autoscale?

Reply:
yeah, we should be able to tell if it is a static or dynamic cluster.

Reply:
any pointers there?